### DISCUSSION TITLE
Misc devices > rtl_433 > mqtt > Home Assistant sensor

### DISCUSSION THREAD

**Discussion by jhj1974** [merbanan/rtl_433#2766] [Q&A] [answered]
I am running rtl_433 without any protocol specified and receiveing various sensors that I would like to integrate in Home Assistant. As most sensors received has multiple data (e.g. temperature, humidity, etc.) I would like it to show up as one sensor with multiple attributes in Home Assistant. Apparently, the Home Assistant MQTT Sensor template relies on the state topic, or at least the documentation focuses on the word state. But rtl_433 does not appear to ever send anything with the state topic even it informs that the state topic is configured. Has anyone sucessfully configured both rtl_433 and the HA MQTT Sensor template to receive for example the Cotech weather station shown below into one HA device?

From MQTT Explorer:
![image](https://github.com/merbanan/rtl_433/assets/96663288/2c5a95c0-fcf2-4ce6-8888-12519ebc1765)

Log from rtl_433:
![image](https://github.com/merbanan/rtl_433/assets/96663288/33d490e8-c65f-482e-83b2-d4de676d5146)

This is my MQTT Sensor so far:
![image](https://github.com/merbanan/rtl_433/assets/96663288/ed0016f6-cbf9-4b6a-98fd-489a7a52adc3)

**Comment by gdt** [comment, marked as answer]
This is not about rtl_433 but you should be clear on main/diagnostic/attribute in the HA sensor model.    Look at rtl_433's examples dir, specifically at `rtl_433_mqtt_relay.py` and `rtl_433_mqtt_hass.py`, which together post mqtt and then json to the autoconfig topic to instantiate devices and sensors.  Not saying it will do what you want, but it is perhaps close.

**Reply by jhj1974** [reply]
Tank you for pointing me to the rtl_433_mqtt_hass.py. That was really the answer.
